### PR TITLE
Replace rand with oorandom

### PR DESCRIPTION
**Pull request by matklad**
bors r+
🤖

### CODE REVIEW

File: crates/rust-analyzer/src/cli/analysis_stats.rs [hunk 1]
```diff
@@ -306,3 +314,10 @@ pub fn analysis_stats(
 
     Ok(())
 }
+
+fn shuffle<T>(rng: &mut Rand32, slice: &mut [T]) {
+    for i in (1..slice.len()).rev() {
+        let idx = rng.rand_range(0..i as u32) as usize;
+        slice.swap(idx, i - 1)
```
Comment (lnicola) [inline crates/rust-analyzer/src/cli/analysis_stats.rs:321]:
```suggestion
        slice.swap(idx, i)
```

**Comment by jens1o** [comment]
What is the advantage of this crate over the more established `rand` crate?

**Comment by lnicola** [comment]
Fewer dependencies, and it's only used for minor features, so it doesn't matter too much.

**Comment by matklad** [comment]
Far fewer deps & not having OS randomness (which is a feature)

**Pull request merged**